### DISCUSSION TITLE
Limiting filesystem access

### DISCUSSION THREAD

**Discussion by rkennedy-mode** [duckdb/duckdb#10000] [Q&A]
Is there a way to list all of the filesystems that DuckDB knows about? I want to be able to disable everything except for LocalFileSystem without assuming I know the names of all of the filesystems. So being able to list the filesystems, remove LocalFileSystem from the list, and then pass that list to the disable_filesystems configuration would be handy. Alternatively, having a configuration setting like `enabled_filesystems` as an inverse of disabled_filesystems could work better. Although, it does create a potential for confusion if someone inadvertently sets both enabled_filesystems and disabled_filesystems in a conflicting manner.

Additionally, has there been any thought with LocalFileSystem (or any filesystem, really) to limit access? In my use case I'd like to be able to specify that LocalFileSystem should be allowed but only for files located under `/some/given/path/` (resolving symlinks in the process so they can't be used to reach beyond the sandbox). I didn't see an easy way to make this happen with the configuration options given. Apologies if it can be achieved today.

**Comment by rkennedy-mode** [comment]
@hannes @Mytherin is this something that's been considered? If not, is this something that DuckDB might consider? We're trying to assess how we might sufficiently lock down a multi-tenant service and as part of that we'd like to know if DuckDB would be amenable to such additions before we go about determining how we'd get those changes.

**Reply by Mytherin** [reply]
Yes, more granular access controls are on our long-term roadmap.

**Comment by tarasyarema** [comment]
Related to https://github.com/duckdb/duckdb/discussions/11778
